### ISSUE TITLE
Eval bug: gemma-3n crash when using HIP

### ISSUE THREAD

**Issue by vhqtvn** [ggml-org/llama.cpp#14448]
### Name and Version

ggml_cuda_init: GGML_CUDA_FORCE_MMQ:    no
ggml_cuda_init: GGML_CUDA_FORCE_CUBLAS: no
ggml_cuda_init: found 1 ROCm devices:
  Device 0: AMD Radeon 890M Graphics, gfx1102 (0x1102), VMM: no, Wave Size: 32
version: 5775 (bd9c981d7)
built with clang version 19.0.0git (/srcdest/rocm-llvm c87081df219c42dc27c5b6d86c0525bc7d01f727) for x86_64-pc-linux-gnu

### Operating systems

Linux

### GGML backends

HIP

### Hardware

AMD Radeon 890M Graphics

### Models

Gemma 3n

### Problem description & steps to reproduce

Running all gemma-3n models works well when using cpu, using HIP result in same crash.

Quick debugging show that the array is all nan:

```
(gdb) p cur_p->data[0].p
$8 = nan(0x400000)
(gdb) p cur_p->data[1].p
$9 = nan(0x400000)
(gdb) p cur_p->data[2].p
$10 = nan(0x400000)
```

### First Bad Commit

$ llama-cli -hf unsloth/gemma-3n-E4B-it-GGUF -co -c 0 -fa -ngl 1000
ggml_cuda_init: GGML_CUDA_FORCE_MMQ:    no
ggml_cuda_init: GGML_CUDA_FORCE_CUBLAS: no
ggml_cuda_init: found 1 ROCm devices:
  Device 0: AMD Radeon 890M Graphics, gfx1102 (0x1102), VMM: no, Wave Size: 32
curl_perform_with_retry: HEAD https://huggingface.co/unsloth/gemma-3n-E4B-it-GGUF/resolve/main/gemma-3n-E4B-it-Q4_K_M.gguf (attempt 1 of 1)...
common_download_file_single: using cached file: /home/user/.cache/llama.cpp/unsloth_gemma-3n-E4B-it-GGUF_gemma-3n-E4B-it-Q4_K_M.gguf
build: 5775 (bd9c981d7) with clang version 19.0.0git (/srcdest/rocm-llvm c87081df219c42dc27c5b6d86c0525bc7d01f727) for x86_64-pc-linux-gnu
main: llama backend init
main: load the model and apply lora adapter, if any
llama_model_load_from_file_impl: using device ROCm0 (AMD Radeon 890M Graphics) - 17672 MiB free
llama_model_loader: loaded meta data with 45 key-value pairs and 847 tensors from /home/user/.cache/llama.cpp/unsloth_gemma-3n-E4B-it-GGUF_gemma-3n-E4B-it-Q4_K_M.gguf (version GGUF V3 (latest))
llama_model_loader: Dumping metadata keys/values. Note: KV overrides do not apply in this output.
llama_model_loader: - kv   0:                       general.architecture str              = gemma3n
llama_model_loader: - kv   1:                               general.type str              = model
llama_model_loader: - kv   2:                               general.name str              = Gemma-3N-E4B-It
llama_model_loader: - kv   3:                           general.finetune str              = 3n-E4B-it
llama_model_loader: - kv   4:                           general.basename str              = Gemma-3N-E4B-It
llama_model_loader: - kv   5:                       general.quantized_by str              = Unsloth
llama_model_loader: - kv   6:                         general.size_label str              = 6.9B
llama_model_loader: - kv   7:                           general.repo_url str              = https://huggingface.co/unsloth
llama_model_loader: - kv   8:                     gemma3n.context_length u32              = 32768
llama_model_loader: - kv   9:                   gemma3n.embedding_length u32              = 2048
llama_model_loader: - kv  10:                        gemma3n.block_count u32              = 35
llama_model_loader: - kv  11:                gemma3n.feed_forward_length arr[i32,35]      = [16384, 16384, 16384, 16384, 16384, 1...
llama_model_loader: - kv  12:               gemma3n.attention.head_count u32              = 8
llama_model_loader: - kv  13:   gemma3n.attention.layer_norm_rms_epsilon f32              = 0.000001
llama_model_loader: - kv  14:               gemma3n.attention.key_length u32              = 256
llama_model_loader: - kv  15:             gemma3n.attention.value_length u32              = 256
llama_model_loader: - kv  16:                     gemma3n.rope.freq_base f32              = 1000000.000000
llama_model_loader: - kv  17:           gemma3n.attention.sliding_window u32              = 512
llama_model_loader: - kv  18:            gemma3n.attention.head_count_kv u32              = 2
llama_model_loader: - kv  19:                   gemma3n.altup.active_idx u32              = 0
llama_model_loader: - kv  20:                   gemma3n.altup.num_inputs u32              = 4
llama_model_loader: - kv  21:   gemma3n.embedding_length_per_layer_input u32              = 256
llama_model_loader: - kv  22:         gemma3n.attention.shared_kv_layers u32              = 15
llama_model_loader: - kv  23:          gemma3n.activation_sparsity_scale arr[f32,35]      = [1.644854, 1.644854, 1.644854, 1.6448...
llama_model_loader: - kv  24:   gemma3n.attention.sliding_window_pattern arr[bool,35]     = [true, true, true, true, false, true,...
llama_model_loader: - kv  25:                    tokenizer.chat_template str              = {{ bos_token }}\n{%- if messages[0]['r...
llama_model_loader: - kv  26:                       tokenizer.ggml.model str              = llama
llama_model_loader: - kv  27:                         tokenizer.ggml.pre str              = default
llama_model_loader: - kv  28:                      tokenizer.ggml.tokens arr[str,262144]  = ["<pad>", "<eos>", "<bos>", "<unk>", ...
llama_model_loader: - kv  29:                      tokenizer.ggml.scores arr[f32,262144]  = [-1000.000000, -1000.000000, -1000.00...
llama_model_loader: - kv  30:                  tokenizer.ggml.token_type arr[i32,262144]  = [3, 3, 3, 3, 3, 4, 3, 3, 3, 3, 3, 3, ...
llama_model_loader: - kv  31:                tokenizer.ggml.bos_token_id u32              = 2
llama_model_loader: - kv  32:                tokenizer.ggml.eos_token_id u32              = 106
llama_model_loader: - kv  33:            tokenizer.ggml.unknown_token_id u32              = 3
llama_model_loader: - kv  34:            tokenizer.ggml.padding_token_id u32              = 0
llama_model_loader: - kv  35:               tokenizer.ggml.add_bos_token bool             = true
llama_model_loader: - kv  36:               tokenizer.ggml.add_sep_token bool             = false
llama_model_loader: - kv  37:               tokenizer.ggml.add_eos_token bool             = false
llama_model_loader: - kv  38:            tokenizer.ggml.add_space_prefix bool             = false
llama_model_loader: - kv  39:               general.quantization_version u32              = 2
llama_model_loader: - kv  40:                          general.file_type u32              = 15
llama_model_loader: - kv  41:                      quantize.imatrix.file str              = gemma-3n-E4B-it-GGUF/imatrix_unsloth.dat
llama_model_loader: - kv  42:                   quantize.imatrix.dataset str              = unsloth_calibration_gemma-3n-E4B-it.txt
llama_model_loader: - kv  43:             quantize.imatrix.entries_count u32              = 459
llama_model_loader: - kv  44:              quantize.imatrix.chunks_count u32              = 1326
llama_model_loader: - type  f32:  422 tensors
llama_model_loader: - type  f16:  108 tensors
llama_model_loader: - type q8_0:    1 tensors
llama_model_loader: - type q4_K:  282 tensors
llama_model_loader: - type q6_K:   34 tensors
print_info: file format = GGUF V3 (latest)
print_info: file type   = Q4_K - Medium
print_info: file size   = 4.90 GiB (6.14 BPW)
load: special tokens cache size = 6414
load: token to piece cache size = 1.9446 MB
print_info: arch             = gemma3n
print_info: vocab_only       = 0
print_info: n_ctx_train      = 32768
print_info: n_embd           = 2048
print_info: n_layer          = 35
print_info: n_head           = 8
print_info: n_head_kv        = 2
print_info: n_rot            = 256
print_info: n_swa            = 512
print_info: is_swa_any       = 1
print_info: n_embd_head_k    = 256
print_info: n_embd_head_v    = 256
print_info: n_gqa            = 4
print_info: n_embd_k_gqa     = 512
print_info: n_embd_v_gqa     = 512
print_info: f_norm_eps       = 0.0e+00
print_info: f_norm_rms_eps   = 1.0e-06
print_info: f_clamp_kqv      = 0.0e+00
print_info: f_max_alibi_bias = 0.0e+00
print_info: f_logit_scale    = 0.0e+00
print_info: f_attn_scale     = 1.0e+00
print_info: n_ff             = 16384
print_info: n_expert         = 0
print_info: n_expert_used    = 0
print_info: causal attn      = 1
print_info: pooling type     = 0
print_info: rope type        = 2
print_info: rope scaling     = linear
print_info: freq_base_train  = 1000000.0
print_info: freq_scale_train = 1
print_info: n_ctx_orig_yarn  = 32768
print_info: rope_finetuned   = unknown
print_info: ssm_d_conv       = 0
print_info: ssm_d_inner      = 0
print_info: ssm_d_state      = 0
print_info: ssm_dt_rank      = 0
print_info: ssm_dt_b_c_rms   = 0
print_info: model type       = E4B
print_info: model params     = 6.87 B
print_info: general.name     = Gemma-3N-E4B-It
print_info: vocab type       = SPM
print_info: n_vocab          = 262144
print_info: n_merges         = 0
print_info: BOS token        = 2 '<bos>'
print_info: EOS token        = 106 '<end_of_turn>'
print_info: EOT token        = 106 '<end_of_turn>'
print_info: UNK token        = 3 '<unk>'
print_info: PAD token        = 0 '<pad>'
print_info: LF token         = 248 '<0x0A>'
print_info: EOG token        = 106 '<end_of_turn>'
print_info: max token length = 48
load_tensors: loading model tensors, this can take a while... (mmap = true)
load_tensors: offloading 35 repeating layers to GPU
load_tensors: offloading output layer to GPU
load_tensors: offloaded 36/36 layers to GPU
load_tensors:        ROCm0 model buffer size =  5022.53 MiB
load_tensors:   CPU_Mapped model buffer size =   288.00 MiB
...............................................
llama_context: constructing llama_context
llama_context: n_seq_max     = 1
llama_context: n_ctx         = 32768
llama_context: n_ctx_per_seq = 32768
llama_context: n_batch       = 2048
llama_context: n_ubatch      = 512
llama_context: causal_attn   = 1
llama_context: flash_attn    = 1
llama_context: freq_base     = 1000000.0
llama_context: freq_scale    = 1
llama_context:  ROCm_Host  output buffer size =     1.00 MiB
llama_kv_cache_unified_iswa: creating non-SWA KV cache, size = 32768 cells
llama_kv_cache_unified:      ROCm0 KV buffer size =   256.00 MiB
llama_kv_cache_unified: size =  256.00 MiB ( 32768 cells,   4 layers,  1 seqs), K (f16):  128.00 MiB, V (f16):  128.00 MiB
llama_kv_cache_unified_iswa: creating     SWA KV cache, size = 1024 cells
llama_kv_cache_unified:      ROCm0 KV buffer size =    32.00 MiB
llama_kv_cache_unified: size =   32.00 MiB (  1024 cells,  16 layers,  1 seqs), K (f16):   16.00 MiB, V (f16):   16.00 MiB
llama_context:      ROCm0 compute buffer size =   520.00 MiB
llama_context:  ROCm_Host compute buffer size =   102.01 MiB
llama_context: graph nodes  = 3143
llama_context: graph splits = 22
common_init_from_params: KV cache shifting is not supported for this context, disabling KV cache shifting
common_init_from_params: setting dry_penalty_last_n to ctx_size = 32768
common_init_from_params: warming up the model with an empty run - please wait ... (--no-warmup to disable)
main: llama threadpool init, n_threads = 12
main: chat template is available, enabling conversation mode (disable it with -no-cnv)
main: chat template example:
<start_of_turn>user
You are a helpful assistant

Hello<end_of_turn>
<start_of_turn>model
Hi there<end_of_turn>
<start_of_turn>user
How are you?<end_of_turn>
<start_of_turn>model

system_info: n_threads = 12 (n_threads_batch = 12) / 24 | ROCm : NO_VMM = 1 | PEER_MAX_BATCH_SIZE = 128 | CPU : OPENMP = 1 | REPACK = 1 |

main: interactive mode on.
sampler seed: 1372900278
sampler params:
        repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
        dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = 32768
        top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, top_n_sigma = -1.000, temp = 0.800
        mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
sampler chain: logits -> logit-bias -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
generate: n_ctx = 32768, n_batch = 2048, n_predict = -1, n_keep = 1

== Running in interactive mode. ==
 - Press Ctrl+C to interject at any time.
 - Press Return to return control to the AI.
 - To return control without starting a new line, end your input with '/'.
 - If you want to submit another line, end your input with '\'.
 - Not using system message. To change it, set a different value via -sys PROMPT

> hi
/usr/lib64/gcc/x86_64-pc-linux-gnu/15.1.1/../../../../include/c++/15.1.1/bits/random.tcc:2668: void std::discrete_distribution<>::param_type::_M_initialize() [_IntType = int]: Assertion '__sum > 0' failed.
zsh: IOT instruction (core dumped)  llama-cli -hf unsloth/gemma-3n-E4B-it-GGUF -co -c 0 -fa -ngl 1000

### Relevant log output

```shell
$ llama-cli -hf unsloth/gemma-3n-E4B-it-GGUF -co -c 0 -fa -ngl 1000
```

**Comment by a1bert01** [comment]
try it with -ngl 35

**Comment by olegshulyakov** [comment]
@vhqtvn Windows build HIP works fine with Gemma 3n

```bat
@echo off

set llama.version=llama-b6031-bin-win-hip-radeon-x64
set llama.cpp=%USERPROFILE%\.llama.cpp\%llama.version%
set port=1234

set alias="gemma3n:e4b"
set model="S:\HuggingFace\bartowski\gemma-3n-E4B-it-GGUF\gemma-3n-E4B-it-Q8_0.gguf"

set ctx-size=32768
set gpu-layers=99

set repeat-penalty=1.0
set temp=1.0
set min-p=0.0
set top-k=64
set top-p=0.95

set cmd=%llama.cpp%\llama-server ^
    --host 0.0.0.0 ^
	--port %port% ^
	--no-webui ^
    --log-file %USERPROFILE%\.llama.cpp\llama-server.log ^
    --gpu-layers %gpu-layers% ^
	--alias %alias% ^
	--model %model% ^
	--ctx-size %ctx-size% ^
	--jinja ^
	--repeat-penalty %repeat-penalty% ^
	--temp %temp% ^
	--top-k %top-k% ^
	--min-p %min-p% ^
	--top-p %top-p%
    
%cmd% --flash-attn

set /p DUMMY=Hit ENTER to continue...
```

**Comment by vhqtvn** [comment]
@olegshulyakov AMD drivers on windows is far better. It also support NPU using lemonade, which is still a delayed plan on Linux.

Just tried again, still crash on latest version.

**Comment by olegshulyakov** [comment]
The issue is still reproducable ob the latest **Vulkan** build 5e6229a8 (6178)

```
llama-bench --model "C:\HuggingFace\ggml-org\gemma-3n-E4B-it-GGUF\gemma-3n-E4B-it-Q8_0.gguf" --n-gpu-layers 99 --n-prompt 512,2048,8192 --n-gen 128,256 --main-gpu 0 --split-mode none --mmap 0 --flash-attn 1 

load_backend: loaded RPC backend from C:\Users\oleg\.llama.cpp\llama-b6178-bin-win-vulkan-x64\ggml-rpc.dll
ggml_vulkan: Found 1 Vulkan devices:
ggml_vulkan: 0 = AMD Radeon RX 7800 XT (AMD proprietary driver) | uma: 0 | fp16: 1 | bf16: 1 | warp size: 64 | shared memory: 32768 | int dot: 1 | matrix cores: KHR_coopmat
load_backend: loaded Vulkan backend from C:\Users\oleg\.llama.cpp\llama-b6178-bin-win-vulkan-x64\ggml-vulkan.dll
load_backend: loaded CPU backend from C:\Users\oleg\.llama.cpp\llama-b6178-bin-win-vulkan-x64\ggml-cpu-icelake.dll
```
| model                          |       size |     params | backend    | ngl |    sm | fa | mmap |            test |                  t/s |
| ------------------------------ | ---------: | ---------: | ---------- | --: | ----: | -: | ---: | --------------: | -------------------: |

```
ggml_vulkan: Device memory allocation of size 2495610880 failed.
ggml_vulkan: Requested buffer size exceeds device memory allocation limit: ErrorOutOfDeviceMemory
main: error: failed to load model 'C:\HuggingFace\ggml-org\gemma-3n-E4B-it-GGUF\gemma-3n-E4B-it-Q8_0.gguf'
```
